src/data_structure/circular_queue.rs

```diff
@@ -0,0 +1,105 @@
+pub struct Queue<T> {
+    data: Vec<T>,
+    index: usize,
+    front: usize,
+    capacity: usize,
+}
+
+impl<T> Queue<T> {
+    // Returns a new Queue object with given capacity
+    pub fn with_capacity(capacity: usize) -> Self {
+        Self {
+            data: Vec::with_capacity(capacity),
+            index: 0,
+            front: 0,
+            capacity,
+        }
+    }
+
+    // Adds the element to the circular queue
+    // Once capacity is reached, new items will overwrite old ones.
+    pub fn enqueue(&mut self, e: T) {
+        if self.len() < self.capacity() {
```

Comment:
We can save a call here, since we can directly use `self.capacity` :)

Reply:
This is necessary as pushing into a newly created vec with capacity will give error. 
This checks helps to check whether we have/had an element at that position, or we need to push a new element.

Reply:
I was just saying if we can do `self.capacity` in place of `self.capacity()` since the function return this same thing.

Reply:
OK, got it. Instead of a function call, use the value from `self` directly. 
Sorry misunderstanding on my part. 😞